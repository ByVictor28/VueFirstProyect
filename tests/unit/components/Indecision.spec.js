import { shallowMount } from "@vue/test-utils"

import Indecision from "@/components/Indecision"
describe("Indecision component", () => {
  let wrapper
  let clgSpy

  global.fetch = jest.fn(()=> Promise.resolve( {
    json: ()=>Promise.resolve({
      "answer":"yes",
      "forsed":false,
      "image":"https://yesno.wtf/assets/yes/2.gift"
    })
  } ) )

  beforeEach(()=> {
    wrapper = shallowMount(Indecision);
    clgSpy =jest.spyOn(console,"log")
  })

  afterEach(()=>{
    jest.clearAllMocks()
  })

  test("snapshot match", () => {
    expect(wrapper.html()).toMatchSnapshot();
  })

  test("type shoud not make a request",async ()=>{

    const getAnswerSpy = jest.spyOn(wrapper.vm, "getAnswer")
    
    const input = wrapper.find("input")
    await input.setValue("Hola munod")
    
    expect(clgSpy).toHaveBeenCalledTimes(1)
    expect(getAnswerSpy).toHaveBeenCalledTimes(0)
  })
  
  test("type '?' should make a request",async ()=>{
    const getAnswerSpy = jest.spyOn(wrapper.vm,"getAnswer")
    const input = wrapper.find("input")
    await input.setValue("Hola?")
    expect(clgSpy).toHaveBeenCalledTimes(1)
    expect(getAnswerSpy).toHaveBeenCalledTimes(1)
  })
  
  test("test getAnswer - Correct API result",async()=>{
    console.log(wrapper.vm)
    await wrapper.vm.getAnswer()
      // expect(wrapper.vm.image).toBe("https://yesno.wtf/assets/yes/2.gift")
    expect(wrapper.vm.answer).toBe("yes")

  })

  test("test getAnswer - Fail API result",async ()=>{
    
    fetch.mockImplementationOnce(()=>Promise.reject("Error"))
    await wrapper.vm.getAnswer()
    expect(wrapper.vm.answer).toBe("Error")
  })


})