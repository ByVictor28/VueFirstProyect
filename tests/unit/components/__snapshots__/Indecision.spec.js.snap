// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision component snapshot match 1`] = `
<img src="https://via.placeholder.com/250" alt="background meme">
<div class="bg-dark"></div>
<div class="container"><input type="text" placeholder="Ask me something">
  <p>Remember finish your question with a (?)</p>
  <!--v-if-->
</div>
`;
